es7/2017-01/jan-25.md

```diff
@@ -0,0 +1,1631 @@
+
+BE: It's not bitwise I'm worried about. What about divide by zero?
+
+WH: That doesn't change between 64 bit and bignums. What happens when you divide by zero with uint64's?
+
+BE: it does because 64 bit hardware exists on many computers that run JavaScript
+
+MM: What does the hardware do?
+
+??: It wraps, usually
+
+MM: presumably your'e going to throw an exception.
+
+BE: maybe i should reconsider if it would be more work. let's not worry about the champion right now. i just want to confirm we don't have a split committee.
+
+AWB: There are two aspects to this, one is the spec time, the other is time to implement. Particularly, time to implement across all the major implementations. Do people actually think they can manage to implement BigInts? What are the differences?
+
+FH: Shouldn't we value users over implementability?
+
+AWB: Where there is a usecase for this, and there are usecases, it's becoming a relatively critical need, this is not something we should be looking for 5 years from now. If we said it was bignums, when can we reasonably expect that implementations would have produced viable performant implementations in browsers?
+
+AK: From the V8 side, we got momentum internally on BigInt because the effrot required seems similar for what people would expect for Int64.
+
+AWB: What do other engines think?
+
+MS: Same for JSC
+
+BE: Just to be clear, do you mean expectations for users on int32...
+
+AK: I don't know what the details of the staging plan...
+
+BE: Does that bring up signed vs unsigned? 
+
+DE: bigints are always signed
+
+??: It's only interesting when shifting
+
+BT: Do you think a general value types is in the cards for JS?
+
+BE: When we got IBM to back off on decimal as a hardcode extension. WE used the same argument that WH cited, value types. Decimal is still out there. I've talked to some people lately that have a need for decimals.
+
+BE: Value types are still on our agenda and we've done some work that everyone agrees with that is constructive. There are all sorts of possible uses, like for the CSSOM; also custom literal suffixes, etc.
+
+BE: Let's talk about this constant rumor that WASM will replace JS. It may happen but not for a long time. We'll have pressure for new value types. Everyone will be happy in number-land until they aren't, and then maybe they'll use bigints when they are available? To me, it almost doesn't matter. We've done enough work on Value types, pressure from the edges is big enough. I'd like use to get there so we don't have to keep hard coding. When I extended the spec, I future-proofed a meta protocol for value types. Realm specific auto wrapping of primitives in objects, that's important. I'm not going to summarize all the learnings we've had. Its useful and could be fruitful. Having been around this current wheel so long it's
```

Comment:
> &lt;lost it&gt;

This will be rendered as HTML. If you want to emphasize in markdown, please use appropriate underscore or bold syntax. 

This must be fixed wherever it appears in these notes.